moveit_ros/planning_interface/planning_scene_interface/src/planning_scene_interface.cpp

```diff
@@ -49,7 +49,8 @@ static const rclcpp::Logger LOGGER = rclcpp::get_logger("moveit_ros.planning_sce
 class PlanningSceneInterface::PlanningSceneInterfaceImpl
 {
 public:
-  explicit PlanningSceneInterfaceImpl(const std::string& ns = "", bool wait = true) : node_(new rclcpp::Node(ns))
+  explicit PlanningSceneInterfaceImpl(const std::string& ns = "", bool wait = true)
+    : node_(new rclcpp::Node("planning_scene_interface_" + std::to_string(reinterpret_cast<std::size_t>(this)), ns))
```

Comment:
What's with this node name? I still think that this will clash with the application node if there is one. Shouldn't we rather pass through a node with default initialization or create a "private" node as for the PlanningSceneMonitor? There we use ros args to set the node name since it's not applied otherwise. Also, there doesn't seem to be an executor anywhere, but that's probably not required as we are spinning the service calls explicitly.

Reply:
Super sorry, I forgot about the node name bug which needs an explicit remapping

> Also, there doesn't seem to be an executor anywhere, but that's probably not required as we are spinning the service calls explicitly.

We don't need an executor here, we do spin the node only after sending a request or publishing a message